chore: sync powerof3/CommonLibSSE + CommonlibVR

langfod opened the pull request:
attempted merge of recent changes from and powerof3/CommonLibSSE + alandtse/CommonLibVR 

- expanded Relocation.test.cpp with versions mentioned in code 6.29,1130,1170,1179

Initial testing has not shown issues. 

Needs thorough review.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Pathfinding subsystem, shader accumulation, and new UI input handlers added.

* **Improvements**
  * Expanded menu/asset metadata and new animation/face-gen and weapon accessors; improved memory/router/system interfaces and formatter behavior for cleaner text output.

* **Bug Fixes**
  * Corrected relocation write size for integral types.

* **Tests / CI**
  * Added versioned test data, a new test preset, and updated CI workflow and test coverage.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

RafearTheModder committed:
fix: fix check for successful ID mapping (#69)

alandtse committed:
Merge branch 'dev' of https://github.com/powerof3/CommonLibSSE into vr

qudix committed:
feat: `REX`

powerof3 committed:
Add `TaskQueueInterface::QueueUpdateNiObject`

powerof3 committed:
Add `ShadowSceneNode::Add/RemoveLight`

powerof3 committed:
RE `WeaponEnchantmentController`

powerof3 committed:
Add `TESObjectREFR::GetEnchantment`

powerof3 committed:
Update flags

powerof3 committed:
Add `NiPoint3` friend operators

FlayaN committed:
maintenance

FlayaN committed:
feat: convert stl::enumeration to REX::EnumSet

FlayaN committed:
Merge branch 'dev' of https://github.com/powerof3/CommonLibSSE into vr

FlayaN committed:
fix: vr specific changes with merge

alandtse committed:
chore: merge clibsse into vr

FlayaN committed:
fix: fix ActorMagicCaster::Update offset (#75)

hakasapl committed:
fix: AE relocation ID for BGSFootstepManager::GetSingleton() (#76)

alandtse committed:
cI: fix clang-format

alandtse committed:
maintenance

FlayaN committed:
Merge remote-tracking branch 'po3/dev' into vr

alandtse committed:
Merge pull request #77 from FlayaN/vr

chore: merge clibsse into vr

alandtse committed:
Merge branch 'dev' of https://github.com/powerof3/CommonLibSSE into vr

alandtse committed:
fix: sync Actor::Unk_117 with flat

Based off neighboring vfuncs

alandtse committed:
maintenance

shuc1 committed:
Update ActorValueList.h

fix: correct misuse of fmt

powerof3 committed:
Merge pull request #165 from shuc1/dev

Fix: avoid using fmt when "__cpp_lib_format" is defined

powerof3 committed:
Update

powerof3 committed:
maintenance

powerof3 committed:
Update `InterfaceStrings` for 1.6.1170

powerof3 committed:
BSScaleformManager - BSScaleformTranslator

powerof3 committed:
Make `TESForm::GetLocalFormID` const

zax-ftw committed:
Add ActorValue related fixes

powerof3 committed:
Merge pull request #167 from zax-ftw/pr-5

ActorValue related fixes and improvements

powerof3 committed:
maintenance

powerof3 committed:
Add `MenuTopicManager::IsCurrentSpeaker`

powerof3 committed:
Fix compile errors

powerof3 committed:
RE `HighActorCuller`

zax-ftw committed:
Refactor ActorValueList methods

powerof3 committed:
Merge pull request #168 from zax-ftw/pr-6

Refactor ActorValueList methods

ArranzCNL committed:
Version.h consistency between NG.

ArranzCNL committed:
CharEvent.h - keyCode naming consistency.

PCH.h - numbers missing.

powerof3 committed:
Merge pull request #169 from ArranzCNL/dev

Consistency issues between NG fork.

alandtse committed:
Merge branch 'dev' of https://github.com/powerof3/CommonLibSSE into vr

alandtse committed:
maintenance

adya committed:
Added index support for BSSimpleList. Fixed cleared list still considered non-empty.

Also added a few handful methods, like getting size() or back() of the list.

powerof3 committed:
Merge pull request #170 from adya/dev

Improvemenrts for BSSimpleList

powerof3 committed:
maintenance

alandtse committed:
feat: add vr HUD_MESSAGE_TYPE

alandtse committed:
maintenance

Acook1e committed:
fix xmake-rules.lua

Rename SKSE_RUNTIME to make compatibility version.h

Acook1e committed:
Update xmake-rules.lua

powerof3 committed:
Merge pull request #171 from Acook1e/patch-1

fix xmake-rules.lua

aers committed:
fix generic param in REL::Relocation::write

powerof3 committed:
Merge pull request #172 from aers/fix-write

fix generic param in REL::Relocation::write

powerof3 committed:
update

powerof3 committed:
maintenance

powerof3 committed:
Update `Hazard::Flags`

powerof3 committed:
Update `Hazard::Flags` again

powerof3 committed:
RE `Explosion`

powerof3 committed:
Add `AIProcess::GetCurrentWeapon`

powerof3 committed:
Add `WeaponEnchantmentController::SetAttachRoot`

powerof3 committed:
Add `AIProcess::GetTorchNode`

wbourne0 committed:
Fix `BSTPointerAndFlags` and `GHashSetBase`

powerof3 committed:
Merge pull request #173 from wbourne0/fix-invalid-names

Fix `BSTPointerAndFlags` and `GHashSetBase`

KrisV-777 committed:
Add facegenkeyframe class definitions

Backporter committed:
fix: BSSmallBlockAllocator.h (#88)

alandtse committed:
maintenance

KrisV-777 committed:
use RelocationID over Offset class & add missing include

KrisV-777 committed:
forward BSFaceGenAnimationData data from Mfg Fix NG

aers committed:
add BSShaderAccumulator

powerof3 committed:
Merge pull request #174 from KrisV-777/dev

Forward face animation data from MfgFix NG

aers committed:
updates

aers committed:
add a little more than rendering classes

powerof3 committed:
Merge pull request #175 from aers/add-renderer-classes

add BSShaderAccumulator and a bunch of pathing stuff

powerof3 committed:
maintenance

Vermunds committed:
Add MenuEventHandlers

powerof3 committed:
Merge pull request #176 from Vermunds/AddMenuEventHandlers

Add missing MenuEventHandlers + fix others

powerof3 committed:
RE `BGSMoviePlayer`

aers committed:
havok memory stuff

aers committed:
implement functions

aers committed:
reformat

powerof3 committed:
Merge pull request #177 from aers/havok

some havok memory stuff

powerof3 committed:
Add missing menu vtables

aers committed:
some ni controller related things

aers committed:
maintenance

powerof3 committed:
Merge pull request #178 from aers/ni-stuff

some ni controller related things

FlayaN committed:
Merge branch 'dev' of https://github.com/powerof3/CommonLibSSE into vr

FlayaN committed:
maintenance

alandtse committed:
Merge pull request #90 from FlayaN/vr

chore: update to latest commonlibsse

langfod committed:
Guard against  fmt v11+ compatibility for enum formatters

Update fmt::formatter specializations to inherit from formatter<string_view>
for compatibility with fmt v10-12. This resolves compilation errors when users
build with fmt v12 while maintaining backward compatibility with v10 and v11.

Changes:
- FormType, ActorValue, EffectArchetype, COL_LAYER, MATERIAL_ID formatters
  now inherit from formatter<string_view>
- REL::Version formatter now inherits from formatter<basic_string_view>
- Removed redundant parse() methods and simplified format() implementations

Fixes enum formatting errors with fmt v12's stricter type requirements.

langfod committed:
add the RE:: namespace qualifier for consistency

- add a flatrim test preset

langfod committed:
Type safety update to Version.h

- should resolve possible issues with CharT or wchar_t

langfod committed:
remove unused CharT template parameter for consistency

langfod committed:
Align Version formatters with enum formatter pattern

Use formatter<std::string_view> inheritance and const reference parameters to match FormTypes.h pattern.

langfod committed:
revert previous as Enum and Version strings do differ

- Enums use FormTypeToString(): returns std::string_view
- Version uses string(): returns std::string (not string_view)
- std::string and std::string_view are different return types
- formatter<std::string, CharT> appears to be the correct base for formatting an owned string with character type conversion

langfod committed:
update std::formater to handle wchar_t

- update std::formater to handle wchar_t with explicit string conversion
- while not common, wide-character formatting could occur in MessageBoxW, translation packages or debuggers
- add to unit test a check for wchar_t conversion

langfod committed:
Merge upstream/vr (PR #90) into ng_update - sync with latest commonlibsse changes

langfod committed:
fix: resolve post-merge compilation issues

- Add missing ClearDeathState() declaration to Actor.h
- Fix ShaderAccumulator StartAccumulating signature to match NiAccumulator base class
- Fix MagicTarget::HasMagicEffectWithKeyword signature mismatch

langfod committed:
docs: improve wchar_t formatter comment clarity

langfod committed:
test: update VR address library test data to v0.194.0

Updated from https://github.com/alandtse/skyrim_vr_address_library/releases/tag/v0.194.0

The old test data had:
- 'v0.31.0' with 'v' prefix causing Version parser to fail
- 12,448 entries

The new test data has:
- '0.194.0' without 'v' prefix (correct format)
- 13,502 entries

This fixes the Module/SupportsSkyrimVR test failure.

langfod committed:
test: expand AE address library test coverage

Add tests for additional AE versions with structural change breakpoints:
- 1.6.629: Major struct layout changes (Actor, ExtraDataList)
- 1.6.1130: BGSSaveLoadManager and Script changes
- 1.6.1170: Latest Steam AE (RUNTIME_SSE_LATEST_AE)
- 1.6.1179: Latest GOG AE

Removed redundant address library files:
- SE versions 1.5.3 through 1.5.80 (1.5.97 is the baseline)
- AE versions 1.6.317-1.6.342, 1.6.640, 1.6.659, 1.6.1170.1

Test coverage now includes 7 address library files covering all key
runtime versions: SE 1.5.97, VR 1.4.15, and AE 353/629/1130/1170/1179.

langfod committed:
added missing includes and forward declarations

alandtse reviewed (commented):
Thanks. Please see if any of the AI comments are right and address/resolve. 

We don't use SKYRIMVR as a define for ng. It's ENABLE_SKYRIM_VR I believe. Regardless if it's in a member structure that would require a runtime getter. 

I thought I saw an function edit to PlayerCharacter or Actor. Those need to make sure it's propagated for all runtimes. Git merge only does it in one place and we need to do the test manually. I may be able to look in the next few days.

langfod commented on include/RE/N/NativeLatentFunction.h:
```diff
@@ -104,7 +104,7 @@ namespace RE
 		void IVirtualMachine::ReturnLatentResult(VMStackID a_stackID, V a_result)
 		{
-			auto var = RE::BSScript::Variable();
+			auto var = RE::BSScript::Variable::Variable();
```
This was actually put in place in 
https://github.com/alandtse/CommonLibVR/commit/65b9e1981d31e1afe1f01c49579c2391d0d5806e#diff-0c1910c03b936508a3af643c52eab748ac5dc636e39524b63fa98dadf72aefad

langfod commented on include/RE/N/NiSourceTexture.h:
```diff
@@ -3,17 +3,63 @@
+				std::int32_t    dwSize;               // 00
+				std::int32_t    dwFlags;              // 04
+				std::int32_t    dwHeight;             // 08
+				std::int32_t    dwWidth;              // 0A
```
from:

https://github.com/powerof3/CommonLibSSE/commit/5d0234613063ecc1f455b21356fab768a7c94d5b#diff-9832033c7a70bac3e256e8af7b9a70e8543553461fbf7ca42923370390b55a88

langfod commented on include/RE/N/NiSourceTexture.h:
```diff
@@ -3,17 +3,63 @@
+				std::int32_t    dwPitchOrLinearSize;  // 10
+				std::int32_t    dwDepth;              // 14
+				std::int32_t    dwMipMapCount;        // 18
+				std::int32_t    dwReserved1[11];      // 1A
```
also:
https://github.com/powerof3/CommonLibSSE/commit/5d0234613063ecc1f455b21356fab768a7c94d5b#diff-9832033c7a70bac3e256e8af7b9a70e8543553461fbf7ca42923370390b55a88

langfod commented on include/RE/N/NiSourceTexture.h:
```diff
@@ -3,17 +3,63 @@
+				std::int32_t    dwReserved1[11];      // 1A
+				DDS_PIXELFORMAT ddspf;                // 48
+				std::int32_t    dwCaps;               // 68
+				std::int32_t    dwCaps2;              // 6A
```
also:
https://github.com/powerof3/CommonLibSSE/commit/5d0234613063ecc1f455b21356fab768a7c94d5b#diff-9832033c7a70bac3e256e8af7b9a70e8543553461fbf7ca42923370390b55a88

langfod committed:
Handle EXCLUSIVE_SKYRIM_VR defines.

langfod commented:
@alandtse 
> Thanks. Please see if any of the AI comments are right and address/resolve.
> 
> We don't use SKYRIMVR as a define for ng. It's ENABLE_SKYRIM_VR I believe. Regardless if it's in a member structure that would require a runtime getter.
> 
> I thought I saw an function edit to PlayerCharacter or Actor. Those need to make sure it's propagated for all runtimes. Git merge only does it in one place and we need to do the test manually. I may be able to look in the next few days.

I updated:
ActorMagicCaster.h
BSShaderAccumulator.h
BStimer.h

- please verify

Most of the other comments may be correct but also appear to be pre-existing issues and some from upstream.
They should probably be looked at but probably in a separate update, yes?

alandtse commented:
Probably needs to be fixed upstream.

alandtse committed:
fix: correct BSTimer size and Explosion members

- Remove incorrect VR size assertion in BSTimer
- Fix EXPLOSION_RUNTIME_DATA_CONTENT macro field names
- Hide Explosion members behind ENABLE_SKYRIM_AE guard

Co-authored-by: factory-droid[bot] <138933559+factory-droid[bot]@users.noreply.github.com>

alandtse committed:
ci: enable CI for ng branch and update actions

- Add ng branch to pull_request triggers
- Update actions to latest versions (v2->v4, v10.0->v11)
- Remove deprecated doNotCache vcpkg option

Co-authored-by: factory-droid[bot] <138933559+factory-droid[bot]@users.noreply.github.com>

alandtse committed:
ci: update vcpkg baseline to December 2024

- Update VCPKG_COMMIT_ID from fc3f613 (July 2023) to f5cbc726b (Dec 2024)
- Fixes CMake 3.31 compatibility issues causing CI failures
- Resolves directxmath build errors
- Tested successfully with VR, ALL, and FLATRIM presets

Co-authored-by: factory-droid[bot] <138933559+factory-droid[bot]@users.noreply.github.com>
